Source/Android/jni/AchievementAdapter.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <jni.h>
+
+#include "Common/Event.h"
+#include "Core/AchievementManager.h"
+#include "jni/AndroidCommon/AndroidCommon.h"
+#include "jni/AndroidCommon/IDCache.h"
+
+extern "C" {
+
+JNIEXPORT void JNICALL
+Java_org_dolphinemu_dolphinemu_features_settings_model_AchievementModel_init(JNIEnv* env, jclass)
+{
+  AchievementManager::GetInstance().Init(nullptr);
+}
+
+JNIEXPORT jboolean JNICALL
+Java_org_dolphinemu_dolphinemu_features_settings_model_AchievementModel_login(JNIEnv* env, jclass,
+                                                                              jstring password)
+{
+  auto& instance = AchievementManager::GetInstance();
+  bool success;
+  auto login_complete_event = std::make_shared<Common::Event>();
+  instance.SetUpdateCallback(
+      [&login_complete_event, &success](AchievementManager::UpdatedItems updated_items) {
+        if (!updated_items.all)
```

Comment:
This feels confusing. Could you add something to UpdatedItems to indicate whether the current callback is about a login result or not? There are currently many uses of UpdatedItems that don't set `all` but are about something completely different than logging in, and it seems like you're relying on those never running when logged out, which is fragile.